cmds/cpu/cpu9p_unix_test.go

```diff
@@ -0,0 +1,199 @@
+// Copyright 2018-2019 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+//go:build !windows && !plan9
+// +build !windows,!plan9
+
+package main
+
+import (
+	"io/ioutil"
+	"reflect"
+
+	"path/filepath"
+	"testing"
+
+	"github.com/hugelgupf/p9/p9"
+	"golang.org/x/sys/unix"
+)
+
+func Test9pUnix(t *testing.T) {
+	d := t.TempDir()
+	f := filepath.Join(d, "a")
+	if err := ioutil.WriteFile(f, []byte("hi"), 0666); err != nil {
+		t.Fatalf(`ioutil.WriteFile(%q, "hi", 0666): %v != nil`, f, err)
+	}
+	// First test: do nothing
+	m := p9.AttrMask{
+		Mode:        true,
+		NLink:       true,
+		UID:         true,
+		GID:         true,
+		RDev:        true,
+		ATime:       true,
+		MTime:       true,
+		CTime:       true,
+		INo:         true,
+		Size:        true,
+		Blocks:      true,
+		BTime:       true,
+		Gen:         true,
+		DataVersion: true,
+	}
+
+	c := &cpu9p{
+		path: f,
+	}
+
+	q, gm, ga, err := c.GetAttr(m)
+	if err != nil {
+		t.Fatalf("Getattr: %v != nil", err)
+	}
+	t.Logf("First getattr returns (%v, %v, %v)", q, gm, ga)
+
+	sam := p9.SetAttrMask{
+		Permissions:        false,
+		UID:                false,
+		GID:                false,
+		Size:               false,
+		ATime:              false,
+		MTime:              false,
+		CTime:              false,
+		ATimeNotSystemTime: false,
+		MTimeNotSystemTime: false,
+	}
+	sa := p9.SetAttr{
+		Permissions:      0777,
+		UID:              ga.UID,
+		GID:              ga.GID,
+		Size:             ga.Size + 4,
+		ATimeSeconds:     ga.ATimeSeconds + 22,
+		ATimeNanoSeconds: ga.ATimeNanoSeconds + 33,
```

Comment:
might go over 10^9 ns.  not sure how things will react to that.  could drop a second, and your DeepEqual check will fail later.